src/main/java/com/semmle/jira/addon/LgtmServlet.java

```diff
@@ -150,6 +152,13 @@ void createIssue(
     issueInputParameters.setIssueTypeId(JiraUtils.getLgtmIssueType().getId());
 
     issueInputParameters.addProperty(Constants.LGTM_PAYLOAD_PROPERTY, rawRequest);
+    CustomFieldManager customFieldManager = ComponentAccessor.getCustomFieldManager();
+
+    CustomField customField =
+        customFieldManager.getCustomFieldObjectByName(Constants.CUSTOM_FIELD_NAME);
+
+    issueInputParameters.addCustomFieldValue(customField.getIdAsLong(), config.getKey());
```

Comment:
I think IssueInputParameters simulates the parameters provided by the user through the UI. Perhaps that's why the custom field gets ignored if it is not defined in the issue-create-screen. It should be possible to write the custom field more directly. There is a method `CustomField::updateValue` which looks a little tricky to use, but sounds like the right thing to do.